src/logger/MetricsContext.ts

```diff
@@ -193,4 +201,19 @@ export class MetricsContext {
       this.timestamp,
     );
   }
+
+  /**
+   * Creates an independently flushable context.
+   * Custom dimensions are NOT preserved.
+   */
+  public createCopyWithContextWithoutDimensions(): MetricsContext {
+    return new MetricsContext(
+      this.namespace,
+      Object.assign({}, this.properties),
+      [],
+      this.defaultDimensions,
+      this.shouldUseDefaultDimensions,
+      this.timestamp,
+    );
```

Comment:
@markkuhn probably you can invoke `createCopyWithContextWithoutDimensions()`, get the result and set dimensions to empty array. This will help you to avoid duplicated code here and in second method

Reply:
Good idea. Instead of clearing the dimensions after `createCopyWithContext()`, I've added a `preserveDimensions` parameter to that function.